test/unit/http/request_test.dart

```diff
@@ -1068,6 +1068,31 @@ void _runAutoRetryTestSuiteFor(String name,
         expect(request.autoRetry.numAttempts, equals(2));
       });
 
+      test(
+          'request timeout should be retried by default increaseTimeoutOnRetry',
+          () async {
+        // 1st and 2nd request = hangs until timeout, 3nd request succeeds
+        int c = 0;
+        MockTransports.http.when(requestUri, (request) async {
+          if (++c <= 2) {
+            await Future.delayed(Duration(seconds: 1));
```

Comment:
:hidethepain: any chance we can use [FakeAsync](https://pub.dev/packages/fake_async) for this instead of having a multi-second unit test?

Reply:
The retry should only wait 25ms.  This Future is arbitrarily long.

Reply:
:think: :nice: